src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side must be greater than 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(
+        'Longest side triangle cannot be greater than the sum of the other two',
+      );
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider explaining why the longest side cannot be greater than the sum of the other two sides.